bundles/org.dataflowanalysis.analysis.dfd/src/org/dataflowanalysis/analysis/dfd/DFDConfidentialityAnalysis.java

```diff
@@ -22,12 +24,22 @@ public class DFDConfidentialityAnalysis extends DataFlowConfidentialityAnalysis
     protected final DFDResourceProvider resourceProvider;
     protected final Optional<Class<? extends Plugin>> modelProjectActivator;
     protected final String modelProjectName;
+    protected final Class<? extends TransposeFlowGraphFinder> transposeFlowGraphFinderClass;
+
+    public DFDConfidentialityAnalysis(DFDResourceProvider resourceProvider, Optional<Class<? extends Plugin>> modelProjectActivator,
+            String modelProjectName, Class<? extends TransposeFlowGraphFinder> transposeFlowGraphFinderClass) {
```

Comment:
Instead of referencing a class here, save the transpose flow graph finder as an attribute in the analysis as a final attribute. In the constructor without the passed transpose flow graph finder, create a new instance of a cyclic transpose flow graph finder, else use the provided one

Reply:
We do this to create the finder in `DFDFlowGraphCollection` at `findTransposeFlowGraphs`like the analysis did before. That way we don't have to deal with not yet ready resource providers

Reply:
```
public List<? extends AbstractTransposeFlowGraph> findTransposeFlowGraphs() {
        if (!(this.resourceProvider instanceof DFDResourceProvider dfdResourceProvider)) {
            logger.error("Cannot find transpose flow graphs for non-dfd resource provider");
            throw new IllegalArgumentException();
        }

        if (transposeFlowGraphFinderClass.equals(DFDTransposeFlowGraphFinder.class))
            this.transposeFlowGraphFinder = new DFDTransposeFlowGraphFinder(dfdResourceProvider);
        else
            this.transposeFlowGraphFinder = new DFDCyclicTransposeFlowGraphFinder(dfdResourceProvider);

        return transposeFlowGraphFinder.findTransposeFlowGraphs();
    }
```

Reply:
But resource provider are readied before findTransposeFlowGraphs is called.
Should they have any issues, the analysis will not be run

Reply:
Discussed with Nicolas:
Rather use enum for finder than Finder.class